lib/ansible/module_utils/facts/system/systemd.py

```diff
@@ -0,0 +1,42 @@
+# Get systemd version and features
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+import ansible.module_utils.compat.typing as t
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+from ansible.module_utils.facts.system.service_mgr import ServiceMgrFactCollector
+
+
+class SystemdFactCollector(BaseFactCollector):
+    name = "systemd"
+    _fact_ids = set()  # type: t.Set[str]
+
+    def collect(self, module=None, collected_facts=None):
+        systemctl_bin = module.get_bin_path("systemctl")
+        if systemctl_bin and ServiceMgrFactCollector.is_systemd_managed(module):
+            rc, stdout, stderr = module.run_command(
+                [systemctl_bin, "--version"],
+                check_rc=False,
+            )
+
+            systemd_facts = {}
+            systemd_facts["systemd"] = {}
+            systemd_facts["systemd"]["features"] = str(stdout.split("\n")[1])
+            systemd_facts["systemd"]["version"] = int(stdout.split(" ")[1])
```

Comment:
If a system is using `systemd` and the binary is present, any user will be able to get the version by default.

Will catch the `.rc`, but is there an integration image without `systemd` or another way to create these fails?

There is no need to waste your time with these simple things, to be honest.

Reply:
In our test targets freebsd and macos images will lack systemd. Some linux distributions and other OSs will lack systemd, while we do not test those directly, that does not mean our user base does not deal with them. 

As for the fails these will be 'random' things like lack of resources or very exotic configurations. I recommend we guard against these because we've encountered such issues before, its just good defensive coding. We don't have to test all these, though adding a 0 open file limit or similar could be one simple way to do so.